### PR TITLE
Update CODEOWNERS

### PR DESCRIPTION
**Pull request by AThousandShips**
* Added shared responsibilities for documentation and tests
* Made buildsystem manage all build scripts (uniquely)
* Cleaned up unused cases
* Added unmanaged cases

TODO:
* [x] Add shared responsibility for `tests`
* [x] Add some missing module cases (needs confirmation on the allocation)
* [ ] Decide on the way to go with `buildsystem`

Any feedback would be welcome!

My current reasoning is:
* For tests and documentation shared responsibility is important
* For buildsystem that's less relevant, we *could* do it but it'd mean a *lot* of clutter in this file (the documentation part is cluttered as it is, and haven't added tests yet)

I'm not sure how to verify these changes as they don't show up in the branch (because of organization management etc. they don't turn up on my own branch)

### CODE REVIEW

File: .github/CODEOWNERS
Comment (AThousandShips) [inline .github/CODEOWNERS:15]: My bad rearranged when removing, will restore

File: .github/CODEOWNERS
Comment (AThousandShips) [inline .github/CODEOWNERS:153]: Wasn't sure what to assign noise to

File: .github/CODEOWNERS
Comment (dalexeev) [inline .github/CODEOWNERS:21]:
```suggestion
**/doc_classes/                         @godotengine/documentation
```

File: .github/CODEOWNERS
Comment (AThousandShips) [inline .github/CODEOWNERS:125]: Assumed this is where this goes

**Comment by AThousandShips** [comment]
I can eventually look at cleaning up some more general cases, like fixing `scene/2d/physics` to belong to `2d-nodes` and `physics`, but I think I'll try keep this simple, the modules part is relatively trivial to rework, but a larger cleanup would be great to help with allocation

**Comment by akien-mga** [comment]
> Added shared responsibilities for documentation and tests

I'm confused by that concept of shared responsibility, I think it might be a misunderstanding of the `CODEOWNERS` patterns.

The current patterns are not exclusive, but are all taken into account.

So
```
doc_classes/     @godotengine/documentation 
modules/gltf/    @godotengine/import
```
Already implies that a change to `modules/gltf/doc_classes/*.xml` will require a review from both the documentation team and the import team. You can see it in https://github.com/godotengine/godot/pull/89444 for example:

image

With only XML changes.

**Comment by AThousandShips** [comment]
> Already implies that a change to modules/gltf/doc_classes/*.xml will require a review from both the documentation team and the import team. 

It doesn't seem that way from various other testing, at least the owners for those files is not shared, can you confirm it actually works like that? I haven't been able to in my own testing

**Comment by dalexeev** [comment]
See [About code owners - CODEOWNERS syntax](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax):

> If you want to match two or more code owners with the same pattern, all the code owners must be on the same line. If the code owners are not on the same line, the pattern matches only the last mentioned code owner.

**Comment by AThousandShips** [comment]
Take this for example, it modified the docs files in csg but didn't request:
* https://github.com/godotengine/godot/pull/93252

**Comment by akien-mga** [comment]
Hm I see, then I misunderstood how it works and we indeed need an approach like this one to be thorough.

File: .github/CODEOWNERS
Comment (Repiteo) [inline .github/CODEOWNERS:8]:
```suggestion
*                                       @godotengine/buildsystem
```

**Review by Repiteo — COMMENTED** [review]
I think making `buildsystem` the de-facto "default" environment makes the most sense. This file would fall under their jurisdiction, so any cases in the future where something isn't accounted for would alert them & subsequently let them update this file as needed.

~~Also, after reading their [documentation](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-file-location), it looks like this file can be moved to the `.github` folder. Maybe this PR could do that as well to help reduce root bloat, but it isn't strictly necessary.~~

**Comment by AThousandShips** [comment]
Move what file? It *is* right in `.github`?

File: .github/CODEOWNERS
Comment (AThousandShips) [reply .github/CODEOWNERS:8]: Will do!

**Comment by Repiteo** [comment]
Oh, duh, so it is. I must've mistaken `CONTRIBUTING.md` for it at a glance, my b

**Comment by akien-mga** [comment]
We've just been discussing this with @Repiteo, I think we should be able to consider this as ready to merge?
It would help a lot to fix the current missing review-requests for areas which the previous `CODEOWNERS` doesn't handle well.

I wanted to do a quick comparative review with some of the changes I made in https://github.com/godotengine/godot/pull/93393 but haven't found the time yet. I don't think there's any really critical difference though so we're probably good to go with this PR as is.

**Comment by AThousandShips** [comment]
Will do a quick pass over this and fill in some potential misses and otherwise I'll consider this ready, and can go back over it in detail

**Comment by AThousandShips** [comment]
Added the new physics modules to the owners but otherwise I'll leave this for future improvements and cleanups, already having a few cases to add and go over

File: .github/CODEOWNERS
Comment (AThousandShips) [inline .github/CODEOWNERS:89]:
```suggestion
/modules/betsy/                         @godotengine/import
/modules/bmp/                           @godotengine/import
```
Will add after CI is done

File: .github/CODEOWNERS
Comment (AThousandShips) [inline .github/CODEOWNERS:153]: Leaving these for the future, but leaving the spacing for future expansion

**Review by akien-mga — APPROVED** [review]

**Comment by AThousandShips** [comment]
Thank *you*!

**Pull request merged**
